### PR TITLE
fix(react-native): implement `projectConfig` and `dependencyConfig`

### PR DESCRIPTION
**Pull request by tido64**
## Summary:

Implements `projectConfig` and `dependencyConfig` so that `react-native config` populates the `macos` field.

## Test Plan:

`react-native config` should populate the `macos` field. Example:

```
[...]
  "project": {
    "visionos": {
      "sourceDir": "/~/example/visionos",
      "xcodeProject": null,
      "assets": []
    },
    "ios": {
      "sourceDir": "/~/example/ios",
      "xcodeProject": {
        "name": "Example.xcworkspace",
        "path": ".",
        "isWorkspace": true
      },
      "automaticPodsInstallation": false,
      "assets": []
    },
    "android": {
      "sourceDir": "/~/example/android",
      "appName": "app",
      "packageName": "com.microsoft.reacttestapp",
      "applicationId": "com.microsoft.reacttestapp",
      "mainActivity": "com.microsoft.reacttestapp.MainActivity",
      "assets": []
    },
    "macos": {
      "sourceDir": "/~/example/macos",
      "xcodeProject": {
        "name": "Example.xcworkspace",
        "path": ".",
        "isWorkspace": true
      },
      "assets": []
    },
    "windows": null
  }
}
```

### CODE REVIEW

File: packages/react-native/package.json [hunk 1]
```diff
@@ -113,6 +113,7 @@
     "@jest/create-cache-key-function": "^29.6.3",
     "@react-native-community/cli": "14.0.0-alpha.2",
     "@react-native-community/cli-platform-android": "14.0.0-alpha.2",
+    "@react-native-community/cli-platform-apple": "14.0.0-alpha.2",
     "@react-native-community/cli-platform-ios": "14.0.0-alpha.2",
```
Comment (Saadnajmi) [inline packages/react-native/package.json:117]: FWIW I think I delete these entries / these entries are deleted in 0.76

**Comment by Saadnajmi** [comment]
Looks like you just need to run `yarn install`

**Comment by tido64** [comment]
> Looks like you just need to run `yarn install`

Looks like we forgot to update it when we bumped Yarn as well:

image

File: packages/react-native/package.json (hunk 1 above)
Comment (tido64) [reply packages/react-native/package.json:117]: We're going to need `@react-native-community/cli-platform-apple` for `getDependencyConfig` and `getProjectConfig` though. This specific package will have to stay.

**Review by Saadnajmi — APPROVED** [review]
Would also appreciate back porting to 0.74 :)

File: packages/react-native/package.json (hunk 1 above)
Comment (Saadnajmi) [reply packages/react-native/package.json:117]: Spoke offline. This stays.

**Comment by tido64** [comment]
> Would also appreciate back porting to 0.74 :)

- **0.74:** https://github.com/microsoft/react-native-macos/pull/2234
- **0.75:** https://github.com/microsoft/react-native-macos/pull/2235

**Pull request merged**